installer/x86_64/install.sh

```diff
@@ -652,7 +652,7 @@ fi
 # Note: assume that apparmor is supported in the kernel
 demo_grub_entry="$demo_volume_revision_label"
 if [ "$install_env" = "sonic" ]; then
-    old_sonic_menuentry=$(cat /host/grub/grub.cfg | sed "/$running_sonic_revision/,/}/!d")
+    old_sonic_menuentry=$(cat /host/grub/grub.cfg | sed "/'SONiC-OS-${running_sonic_revision}'/,/}/!d")
```

Comment:
Looking at the code of this script I see that menuentry is generated like this:
```
# The build system prepares this script by replacing %%DEMO-TYPE%%
# with "OS" or "DIAG".
demo_type="%%DEMO_TYPE%%"
demo_volume_label="SONiC-${demo_type}"
demo_volume_revision_label="SONiC-${demo_type}-${image_version}"
demo_grub_entry="$demo_volume_revision_label"
....
cat <<EOF >> $grub_cfg
menuentry '$demo_grub_entry' {
```

So, it can be SONiC-OS or SONiC-DIAG theoretically, maybe match like this?:
```bash
 sed "/'SONiC-.*-${running_sonic_revision}'/,/}/!d")
```

However, I am not sure we ever build DIAG type, so suggestion is:

```bash
 sed "/'${demo_volume_label}-${running_sonic_revision}'/,/}/!d")
```

Reply:
Make sense